src/plugins/plugin.filler.js

```diff
@@ -232,47 +233,120 @@ function isDrawable(point) {
 	return point && !point.skip;
 }
 
-function drawArea(ctx, curve0, curve1, len0, len1) {
+function fillPointsSets(ctx, curve0, curve1, len0, len1, area, pointSets) {
 	var i, cx, cy, r;
+	var fillAreaPointsSet = [];
+	var clipAboveAreaPointsSet = [];
+	var clipBelowAreaPointsSet = [];
 
 	if (!len0 || !len1) {
 		return;
 	}
+	clipAboveAreaPointsSet.push({x: curve1[len1 - 1].x, y: area.top});
+	clipBelowAreaPointsSet.push({x: curve0[0].x, y: area.top});
+	clipBelowAreaPointsSet.push(curve0[0]);
 
 	// building first area curve (normal)
-	ctx.moveTo(curve0[0].x, curve0[0].y);
+	fillAreaPointsSet.push(curve0[0]);
 	for (i = 1; i < len0; ++i) {
-		helpers.canvas.lineTo(ctx, curve0[i - 1], curve0[i]);
+		curve0[i].flip = false;
+		fillAreaPointsSet.push(curve0[i]);
+		clipBelowAreaPointsSet.push(curve0[i]);
 	}
 
 	if (curve1[0].angle !== undefined) {
+		pointSets.fill.push(fillAreaPointsSet);
+		var radialSet = [];
 		cx = curve1[0].cx;
 		cy = curve1[0].cy;
 		r = Math.sqrt(Math.pow(curve1[0].x - cx, 2) + Math.pow(curve1[0].y - cy, 2));
 		for (i = len1 - 1; i > 0; --i) {
-			ctx.arc(cx, cy, r, curve1[i].angle, curve1[i - 1].angle, true);
+			radialSet.push({cx: cx, cy: cy, radius: r, startAngle: curve1[i].angle, endAngle: curve1[i - 1].angle});
+		}
+		if (radialSet.length) {
+			pointSets.fill.push(radialSet);
 		}
 		return;
 	}
-
 	// joining the two area curves
-	ctx.lineTo(curve1[len1 - 1].x, curve1[len1 - 1].y);
+	var jointPoint = {};
```

Comment:
this should be declared at the top of the `function` to match existing code style